src/main/java/org/elasticsearch/action/admin/indices/segments/IndicesSegmentsRequest.java

```diff
@@ -37,7 +37,7 @@ public IndicesSegmentsRequest() {
 
     public IndicesSegmentsRequest(String... indices) {
         super(indices);
-        indicesOptions(IndicesOptions.fromOptions(false, false, true, false));
+        indicesOptions(IndicesOptions.fromOptions(false, true, true, false));
```

Comment:
I checked two of the other cat apis for their internal treatment of the IndicesOptions, at least the "_cat/shards", "_cat/count" and "_cat/indices" return without exception when the cluster is empty. They internally use an IndicesStatsRequest which defaults to 'IndicesOptions.strictExpandOpenAndForbidClosed()' and that flavor of options setting has the 'allowNoIndices' set to true like in my proposed changes.

I also checked the current behaviour of the non-cat indices apis "_count", "_stats" which seem to be comparable to me. They internally also build requests (CountRequest, IndicesStatsRequest) which have the IndicesOptions.allowNoIndices flag turned on. "_segments" seems to differ for some reason, but I don't really see why. If there is a reason for the diverging behaviour of both "_segments" and "_cat/segments" there is no real need to change this. Anyone else more knowledgable in that regard?